pyigm/clustering/survey_2d2pcf.py

```diff
@@ -133,34 +178,71 @@ def calc_xi_gg(self, sigma=0, error=None):
+        if auto:
+            nRD = nDR
+        else:
+            nRD = getattr(self,'nR{:s}D{:s}'.format(xa,xb))
 
+        # Calculate me
         s = sigma
-        Wgg, err_W = W3(gf(self.DgDg, s), gf(self.RgRg, s), gf(self.DgRg, s), gf(self.DgRg, s), Ndd=self.nDgDg,
-                        Nrr=self.nRgRg, Ndr=self.nDgRg, Nrd=self.nDgRg)
+        W, err_W = W3(gf(DD(self), s), gf(RR(self), s), gf(DR(self), s), gf(RD(self), s),
+                        nDD, nRR, nDR, nRD)
         #
-        self.xi_gg = Wgg
-        self.xi_gg_err_ls = err_W
 
         # jacknife error
         if error == 'jk':
             err_Wjk = np.zeros((len(self.rbinedges) - 1, len(self.tbinedges) - 1), float)
             psd_val = np.zeros((len(self.rbinedges) - 1, len(self.tbinedges) - 1), float)
             N = len(self.fields)
             for field in self.fields:
-                DgDg_aux = self.DgDg - field.DgDg
-                RgRg_aux = self.RgRg - field.RgRg
-                DgRg_aux = self.DgRg - field.DgRg
-                Wgg_aux, _ = W3(gf(DgDg_aux, s), gf(RgRg_aux, s), gf(DgRg_aux, s), gf(DgRg_aux, s), Ndd=self.nDgDg,
-                                Nrr=self.nRgRg, Ndr=self.nDgRg, Nrd=self.nDgRg)
-                psd_val += N * Wgg - (N - 1) * Wgg_aux
+                DD_aux = DD(self) - DD(field) #self.DgDg - field.DgDg
+                RR_aux = RR(self) - RR(field) #self.RgRg - field.RgRg
+                DR_aux = DR(self) - DR(field) #self.DgRg - field.DgRg
+                if auto:
+                    RD_aux = DR_aux
+                else:
+                    RD_aux = RD(self) - RD(field) #self.DgRg - field.DgRg
+                W_aux, _ = W3(gf(DD_aux, s), gf(RR_aux, s), gf(DR_aux, s), gf(RD_aux, s),
+                                Ndd=nDD, Nrr=nRR, Ndr=nDR, Nrd=nRD)
+                                #Ndd=self.nDgDg, Nrr=self.nRgRg, Ndr=self.nDgRg, Nrd=self.nDgRg)
+                psd_val += N * W - (N - 1) * W_aux
             mean_psd = psd_val / N
             for field in self.fields:
-                DgDg_aux = self.DgDg - field.DgDg
-                RgRg_aux = self.RgRg - field.RgRg
-                DgRg_aux = self.DgRg - field.DgRg
-                Wgg_aux, _ = W3(gf(DgDg_aux, s), gf(RgRg_aux, s), gf(DgRg_aux, s), gf(DgRg_aux, s), Ndd=self.nDgDg,
-                                Nrr=self.nRgRg, Ndr=self.nDgRg, Nrd=self.nDgRg)
-                err_Wjk += (mean_psd - Wgg_aux) ** 2
+                DD_aux = DD(self) - DD(field) #self.DgDg - field.DgDg
+                RR_aux = RR(self) - RR(field) #self.RgRg - field.RgRg
+                DR_aux = DR(self) - DR(field) #self.DgRg - field.DgRg
+                if auto:
+                    RD_aux = DR_aux
+                else:
+                    RD_aux = RD(self) - RD(field) #self.DgRg - field.DgRg
+                W_aux, _ = W3(gf(DD_aux, s), gf(RR_aux, s), gf(DR_aux, s), gf(RD_aux, s),
+                                Ndd=nDD, Nrr=nRR, Ndr=nDR, Nrd=nRD)
```

Comment:
Possible issue here: the new normalizations for each jacknife (or bootstrap) iteration should be with the new pair counts, not the original ones. Do you agree?

Reply:
yes, I agree.  

I haven't been using this method so I am going to put a pdb.set_trace()
here with a warning that the code needs to be properly developed before using.

Same with bootstrap